src/util.js

```diff
@@ -580,24 +580,24 @@ function getNodeKindIncludingLogical(node) {
 }
 
 /**
- * Check if string can safely be converted from double to single quotes, i.e.
+ * Check if string can safely be converted from double to single quotes and vice-versa, i.e.
  *
  * - no embedded variables ("foo $bar")
  * - no linebreaks
  * - no special characters like \n, \t, ...
  * - no octal/hex/unicode characters
  *
- * See http://php.net/manual/en/language.types.string.php#language.types.string.syntax.double
+ * See https://php.net/manual/en/language.types.string.php#language.types.string.syntax.double
  */
 function useDoubleQuote(node, options) {
-  if (node.isDoubleQuote && options.singleQuote) {
+  if (node.isDoubleQuote === options.singleQuote) {
+    // We have a double quote and the user passed singleQuote:true, or the other way around.
     const rawValue = node.raw.slice(node.raw[0] === "b" ? 2 : 1, -1);
-
-    return rawValue.match(
-      /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"/
+    const isComplex = rawValue.match(
+      /\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"|\$/
```

Comment:
I am very afraid of such things, but it is better than we have, so let's merge

Reply:
I didn't even notice this diff when I first looked at the PR :see_no_evil: 
The reason we didn't need to check for `$` when converting double to single quotes is that the parser splits them into `string` and `encapsedpart` nodes, such that the parts with the `dollar` are not even seen by the code that handles `string` ([example](https://loilo.github.io/prettier-php-playground/#N4IgDgTgpgLjCWUIHkwIPYDsDOIBcok8mMA6vACYwAW+AHAAwA0IMAhgEblW14AsLAK7YoAFU648AMzYAbES2zEA5rKgBFQehhR8M+VBZhqYAGpIlWfCACsAOj4gWMCG3iyVAYXQBbH2wAFAAkA-BdBQxAOVwBjKABlGABPNWswbAgAWgAmJxBoAEdBeGgA12V-PTkFEGIRCBgytgq2KoMAXxYoCngYdBQ0eCxJUDZsGDCICM7azDBBCbwQAB4AfmMwAB1Mbc2QKXR0AAIAEg42CD2AbhB2oA)).

The only unwanted side-effect this might introduce is that we wouldn't convert a string like `$ 5.00` even though it is actually "simple" (didn't test this yet, cc @utkarshgupta137)

Reply:
Yeah, that's why I mentioned it in the PR itself. I verified it doesn't affect behavior using AST_COMPARE=1.
For some reason, it doesn't work the other way around. So I had to add this regex too.

Reply:
We can either expand the syntax to \$\w or we can have the parser split single quotes too.

Reply:
It seems that strings containing the `$` sign like `"5 $"` are currently not part of our test suite, so the tests don't show that we actually handle them differently with this change:
- `"5 $"` -> `'5 $'` with single-quote=true (before)
- `"5 $"` -> `"5 $"` with single-quote=true (now)

I think we should try to fix this by adding the official regex for PHP variable names, `^[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*$` (see [here](https://www.php.net/language.variables.basics)). Think that would work @utkarshgupta137?

cc @evilebottnawi

Reply:
This regex is only for variable names. This regex doesn't contain all syntaxes for string parsing.

Reply:
That's right - I meant we have to add it to the regex that we already have. Something like
```
/\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"|\$[a-zA-Z_\x80-\xff][a-zA-Z0-9_\x80-\xff]*/
```

Reply:
Yeah, but it doesn't cover complex string parsing.

Reply:
How do you mean? Can you give an example?

Reply:
https://www.php.net/manual/en/language.types.string.php#language.types.string.parsing
See complex examples.

Reply:
Ah I see, some example of what we'd miss if we just check for variables:
```php
'He drank some juice made of ${juice}s.'
'result: {${getName()}}'
```
Since all the complex expressions seem to contain the `$` character I'm fine with adding just that to our regex when converting from single to double quotes. When converting the other way around, I'd prefer to leave `$` out though, such that we can still convert "5 $" to '5 $' when `single-quote` is set to `true`.

Reply:
Yes but if by chance it says $bill or something like that, then converting from single to double quote will result in an error or change in syntax

Reply:
You mean converting from double to single quotes? That won't happen because they're already split in the parser. And the other way around is caught by the `$` part that you added in the regex.

I'd implement it roughly like this:
```js
const isComplex = 
  (!node.isDoubleQuote && rawValue.includes("$")) || 
  rawValue.match(/\\([$nrtfve]|[xX][0-9a-fA-F]{1,2}|[0-7]{1,3}|u{([0-9a-fA-F]+)})|\r?\n|'|"/)
}
```
Or am I missing something?

Reply:
Sorry, I misunderstood what you said.
This would work. But I personally don't like the idea of things working one way but not the other.
I'll leave the final call to you.

Reply:
Fair point, ist would make ist easier to think about this feature if the same regex is used. 
@evilebottnawi @loilo what’s your take on this?

Reply:
Any updates?

Reply:
Sorry, I was a bit absorbed with other things in the last few days. What I didn't consider before is that the regression mentioned above,

> * `"5 $"` -> `'5 $'` with single-quote=true (before)
> * `"5 $"` -> `"5 $"` with single-quote=true (now)

is actually only affecting users who convert their non-pretty codebase for the first time using `single-quote=true`. Anyone who's using the plugin already and updates to the new release, won't see any regressions because `"5 $"` has already been converted to `'5 $'` and will stay that way.

So I'd stay let's stick with the current implementation :+1: